### DISCUSSION TITLE
GPT-4 with Dall E-3

### DISCUSSION THREAD

**Discussion by proteus-dev** [DeabLabs/cannoli#7] [Ideas]
Hi, 

I try using Cannoli for generating articles (whitepaper style). 
These whitepapers always have three images as assets. 
Currently, I go to open AI's page to access GPT-4 for generating them.

Is/will there be a way to generate images (from) inside Cannolis?

Thanks and keep up the great work! 
Cannoli is already very useful for me!

**Comment by blindmansion** [comment]
There's currently not a way to call DALLE using cannoli, but this is a great idea, and something I'd love to add.

Openai doesn't have a public API for DALLE-3 yet, just DALLE-2. I'll likely wait to implement this feature into cannoli until the new image API comes out, as DALLE-2's quality isn't super useful yet, and the API structure and features will likely be different when DALLE-3 is added.

Thanks, I appreciate it!

**Reply by proteus-dev** [reply]
Fully agree. I used Midjourney 4/5 until recently DALLE-3 was released.

**Reply by proteus-dev** [reply]
Generating images/assets from Cannolis could become very interesting... I could imagine that open AI one day will train DALLE models for mockup and asset generation (maybe they are even on the way already...). This could be especially interesting in the context of the idea in my linked post below.  
https://github.com/DeabLabs/cannoli/discussions/6#discussioncomment-7338082
